tests/ga/test_policy_engine.py

```diff
@@ -15,20 +15,376 @@
+        """
+        When conf flag is set to true and policy file is present at expected location, feature should be enabled.
+        """
+        test_extension = Extension(name=TEST_EXTENSION_NAME)
         with patch('azurelinuxagent.ga.policy.policy_engine.conf.get_extension_policy_enabled', return_value=True):
-            engine = PolicyEngine()
-            self.assertTrue(engine.is_policy_enforcement_enabled(),
-                            msg="Conf flag is set to true so policy enforcement should be enabled.")
+            # Create dummy policy file at the expected path to enable feature.
+            with open(self.custom_policy_path, mode='w') as policy_file:
+                json.dump({}, policy_file, indent=4)
+                policy_file.flush()
+                engine = ExtensionPolicyEngine(test_extension)
+                self.assertTrue(engine.is_policy_enforcement_enabled(),
+                                msg="Conf flag is set to true so policy enforcement should be enabled.")
 
     def test_policy_enforcement_should_be_disabled(self):
-        engine = PolicyEngine()
+        self.patch_conf_flag.stop()  # Turn off the policy feature enablement
+        test_extension = Extension(name=TEST_EXTENSION_NAME)
+        engine = ExtensionPolicyEngine(test_extension)
         self.assertFalse(engine.is_policy_enforcement_enabled(),
                          msg="Conf flag is set to false so policy enforcement should be disabled.")
 
+    def test_should_allow_and_should_not_enforce_signature_for_default_policy(self):
+        """
+        Default policy should allow all extensions and not enforce signature.
+        """
+        test_extension = Extension(name=TEST_EXTENSION_NAME)
+        engine = ExtensionPolicyEngine(test_extension)
+        should_allow = engine.should_allow_extension()
+        self.assertTrue(should_allow, msg="Default policy should allow all extensions.")
+        should_enforce = engine.should_enforce_signature()
+        self.assertFalse(should_enforce, msg="Default policy should not enforce extension signature.")
+
+    def test_should_allow_if_allowListedExtensionsOnly_true_and_extension_in_list(self):
+        """
+        If allowListedExtensionsOnly is true and extension in list, should_allow = True.
+        """
+        test_extension = Extension(name=TEST_EXTENSION_NAME)
+        policy = \
+            {
+                "policyVersion": "0.1.0",
+                "extensionPolicies": {
+                    "allowListedExtensionsOnly": True,
+                    "signatureRequired": False,
+                    "extensions": {
+                        TEST_EXTENSION_NAME: {
+                            "signatureRequired": False,
+                            "signingPolicy": {},
+                            "runtimePolicy": {}
+                        }
+                    }
+                }
+            }
+        with open(self.custom_policy_path, mode='w') as policy_file:
```

Comment:
```
        with open(self.custom_policy_path, mode='w') as policy_file:
            json.dump(policy, policy_file, indent=4)
            policy_file.flush()

```

should probably be refactored into a helper method

Reply:
good point, I added this helper